src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilder.java

```diff
@@ -233,7 +233,7 @@ public Pod build() {
         }
 
         // merge with the yaml fragments
-        Pod pod = combine(template.getYamlsPod(), builder.endSpec().build());
+        Pod pod = PodDecorator.decorateAll(new PodBuilder(combine(template.getYamlsPod(), builder.endSpec().build()))).build();
```

Comment:
Does this cost anything if the list is empty?

Reply:
It's pretty cheap but I refactored to avoid it.